src/utils/apiHandler/historyVideoHandler.js

```diff
@@ -0,0 +1,52 @@
+import axios from "axios"
+import { GET_HISTORY_VIDEOS } from "../../reducer/video/videoListConstant"
+
+const encodedtoken = localStorage.getItem("token");
+const config = {
+  headers: {
+    authorization: encodedtoken,
+  },
+};
+
+export const getHistoryVideos = async(dispatch) => {
+    try {
+        const response = await axios.get("/api/user/history", config)
+        dispatch({type: GET_HISTORY_VIDEOS, payload: response.data.history})
+    }
+    catch(err) {
+        console.log(err)
+    }
+}
+
+export const addToHistoryVideo = async(video, dispatch) => {
+    const data = {
+        video,
+    }
+    try {
+        const response = await axios.post("/api/user/history", data, config)
+        dispatch({type: GET_HISTORY_VIDEOS, payload: response.data.history})
```

Comment:
You should check the `response.status` and then use dispatch.
By this, you can know what status code does the api request has returned.
This helps in better error handling as well.